Dockerfile

```diff
@@ -12,10 +12,12 @@
 #  License for the specific language governing permissions and limitations under
 #  the License.
 
-FROM usdotfhwastoldev/carma-base:develop as deps
-
+ARG DOCKER_ORG="usdotfhwastoldev"
+ARG DOCKER_TAG="develop"
+FROM ${DOCKER_ORG}/carma-base:${DOCKER_TAG} as deps
 
 FROM deps as setup
+ARG GIT_BRANCH="develop" 
```

Comment:
I think I figured it out, this sets it as an ARG, but the checkout.bash line below is referencing it as an unitialized environment variable within the run command I think. You might need an `ENV GIT_BRANCH=${GIT_BRANCH}` here to initialize this as an environment variable as well.

Reply:
I just checked the docker build logs and this works fine https://github.com/usdot-fhwa-stol/novatel_gps_driver/actions/runs/8823563427/job/24224278982#step:6:294. Not sure what I am missing here. It correctly checks  out both repos

Reply:
Never mind. You were looking at the circleci failure. The issue was not in the dockerfile but in the checkout script. Added a default to develop although I am not sure this CI ever ran successfully.

Reply:
I am considering fixing the CircleCI workflow for sonar scanner analysis out of scope.